ACM Digital Library.js

```diff
@@ -1,203 +1,200 @@
-	// As of 10/2019, embedded URL can be HTTP even when page is served via HTTPS proxy
-	if (bibtexURL.startsWith('http:') && doc.location.href.startsWith('https')) {
-		Z.debug("Forcing BibTeX URL to HTTPS")
-		bibtexURL = bibtexURL.replace(/^http:/, 'https:');
-	}
-	Zotero.debug('BibTeX URL: ' + bibtexURL);
-	
-	ZU.doGet(bibtexURL, function (text) {
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
-		translator.setString(text);
-		translator.setHandler("itemDone", function (obj, item) {
-			//get the URL for the pdf fulltext from the metadata
-			var pdfURL = ZU.xpath(doc, '//meta[@name="citation_pdf_url"]/@content')[0];
-			if (pdfURL) {
-				pdfURL = pdfURL.textContent.replace(/dl[.-]acm[.-]org[^\/]*/, "dl.acm.org"); //deproxify URL
-				// As of 10/2019, embedded URL can be HTTP even when page is served via HTTPS proxy
-				if (pdfURL.startsWith('http:') && doc.location.href.startsWith('https')) {
-					Z.debug("Forcing PDF URL to HTTPS")
-					pdfURL = pdfURL.replace(/^http:/, 'https:');
-				}
-				Z.debug("PDF URL: " + pdfURL);
-				item.attachments = [{
-					url: pdfURL,
-					title: "ACM Full Text PDF",
-					mimeType: "application/pdf"
-				}];
-			}
-			
-			//fix DOIs if they're in URL form
-			if (item.DOI) item.DOI = item.DOI.replace(/^.*\/(10\.\d+\/)/, '$1');
-			
-			//Conference Locations shouldn't go int Loc in Archive (nor should anything else)
-			delete item.archiveLocation;
-			
-			// some bibtext contains odd </kwd> tags - remove them
-			for (var i=0; i<item.tags.length; i++) {
-				item.tags[i] = item.tags[i].replace("</kwd>", "");
-			}
-			
-			//full issues of journals/magazines don't have a title
-			if (!item.title && text.includes("issue_date")) {
-				var m = text.match(/issue_date\s*=\s*{(.*)},?/);
-				item.itemType = "book";
-				item.title = item.publicationTitle;
-				if (m) {
-					item.title = item.title + ", " + m[1];
-				}
+	let extractedContext = doc.querySelector('meta[name=pbContext]').content;
+	let doiRegex = /article:article:doi\\:([^;]+)/;
+	let doi = extractedContext.match(doiRegex)[1].toLowerCase();
+	let lookupEndpoint = 'https://dl.acm.org/action/exportCiteProcCitation?targetFile=custom-bibtex&format=bibTex&dois=';
+	ZU.doGet(lookupEndpoint + doi, function (text) {
+		let json = JSON.parse(text);
+		let cslItem = json.items[0][doi];
+		cslItem.type = cslItem.type.toLowerCase().replace('_', '-');
+		
+		// For theses, the advisor is indicated as an editor in CSL which
+		// ZU.itemFromCSLJSON incorrectly extracts as an author.
+		if (cslItem.type == 'thesis') {
+			delete cslItem.editor;
+		}
+		
+		let item = new Zotero.Item();
+		ZU.itemFromCSLJSON(item, cslItem);
+		
+		let abstractElements = doc.querySelectorAll('div.article__abstract p, div.abstractSection p');
+		let abstract = Array.from(abstractElements).map(function (element) {
+			return element.textContent;
+		}).join(' ');
```

Comment:
You can also use the short form
```
map(x => x.textContent)
```
Moreover, I would suggest to use `.join('\n\n')` instead, because you are dealing here with different paragraphs in `p` nodes.

Reply:
I forgot about this notation, thanks